common/arcus-model/platform-client/src/main/java/com/iris/client/impl/netty/HttpRequester.java

```diff
@@ -241,16 +241,18 @@ protected void sendErrorToPending(Throwable cause) {
    
    private class HttpRequestInitializer extends ChannelInitializer<SocketChannel> {
       private final SslContext sslCtx;
-      
-      HttpRequestInitializer(SslContext sslCtx) {
+      private final URI uri;
+
+       HttpRequestInitializer(SslContext sslCtx, URI u) {
          this.sslCtx = sslCtx;
+         this.uri = u;
       }
 
       @Override
       protected void initChannel(SocketChannel ch) throws Exception {
          ChannelPipeline pipeline = ch.pipeline();
          if (sslCtx != null) {
-            pipeline.addLast(sslCtx.newHandler(ch.alloc()));
+            pipeline.addLast(sslCtx.newHandler(ch.alloc(), this.uri.getHost(), this.uri.getPort()));
```

Comment:
Does the URI explicitly have the port in it? It doesn't appear this would default to 443 if the scheme is https:


```java
import java.net.URI;
import java.net.URISyntaxException;

class Main {
  public static void main(String[] args) throws URISyntaxException {
    System.out.println(new URI("https://google.com").getPort());
  }
}
```

Prints the value `-1`

Reply:
looks like the original code accounted for this on line 164. Ill include the port generated there.